src/Algolia.Search.Test/EndToEnd/Index/IndexingTest.cs

```diff
@@ -282,6 +288,27 @@ public async Task MoveIndexTest()
             Assert.True(listIndices.Items.Exists(x => x.Name.Equals(indexDestName)));
             Assert.False(listIndices.Items.Exists(x => x.Name.Equals(_indexMoveName)));
         }
+
+        [Test]
+        [Parallelizable]
+        public async Task IndexOperationsAsyncWithJObjectTest()
```

Comment:
Good test. Could you please also test the case when the `JObject` doesn't have an `objectID`?